src/datatype.c

```diff
@@ -1736,8 +1736,10 @@ JL_DLLEXPORT int jl_field_index(jl_datatype_t *t, jl_sym_t *fld, int err)
             }
         }
     }
-    if (err)
+    if (err == 1)
         jl_has_no_field_error(t, fld);
+    if (err == 2)
```

Comment:
temporary change. Trying to use existing signature. Avoiding more untrackable changes.